src/modules/alwaysontop/AlwaysOnTop/FrameDrawer.cpp

```diff
@@ -183,13 +183,21 @@ D2D1_COLOR_F FrameDrawer::ConvertColor(COLORREF color)
 
 D2D1_ROUNDED_RECT FrameDrawer::ConvertRect(RECT rect, int thickness, int radius)
 {
-    auto d2d1Rect = D2D1::RectF((float)rect.left + thickness, (float)rect.top + thickness, (float)rect.right - thickness, (float)rect.bottom - thickness);
-    return D2D1::RoundedRect(d2d1Rect, (float)radius, (float)radius);
+    float halfThickness = thickness / 2.0f;
+    auto d2d1Rect = D2D1::RectF((float)rect.left + halfThickness + 1, 
+        (float)rect.top + halfThickness + 1, 
+        (float)rect.right - halfThickness - 1, 
+        (float)rect.bottom - halfThickness - 1);
+    return D2D1::RoundedRect(d2d1Rect, radius * 2.0f, radius * 2.0f);
 }
 
 D2D1_RECT_F FrameDrawer::ConvertRect(RECT rect, int thickness)
```

Comment:
I see you don't want to use one function, even is radius == 0 ?

Reply:
Sorry, I think we would better leave it as it is or make it optional. I'm afraid some users won't like it and will start opening issues.